src/libraries/System.Security.Cryptography/tests/X509Certificates/PfxTests.cs

```diff
@@ -250,6 +250,22 @@ public static void ECDHPrivateKeyProperty_WindowsPfx()
             }
         }
 
+        [Fact]
+        public static void ECDHPrivateKey_PfxKeyIsEcdsaConstrained()
+        {
+            using (ECDsa ca = ECDsa.Create(ECCurve.NamedCurves.nistP256))
+            {
+                CertificateRequest req = new("CN=potatos", ca, HashAlgorithmName.SHA256);
+
+                using (X509Certificate2 cert = req.CreateSelfSigned(DateTimeOffset.Now, DateTimeOffset.Now.AddDays(3)))
+                {
+                    X509Certificate2 loaded = new X509Certificate2(cert.Export(X509ContentType.Pkcs12, "carrots"), "carrots");
+                    Assert.Null(loaded.GetECDiffieHellmanPrivateKey());
+                    Assert.NotNull(loaded.GetECDiffieHellmanPublicKey());
```

Comment:
OK, so this is "nothing about the certificate says it _can't_ be used for ECDH", but come private key time we find out that Windows has stored the key as signature-only (e.g. alggroup==ECDSA)?

Reply:
Right. In this case the cert itself has no keyUsage, so it's all good for the public key. The private key though has an "ECDSA key type" in the PKCS12 bag attribute, so that attaches "ECDSA-ness" to the private key.

Reply:
Is it worth adding either of

```C#
Assert.True(loaded.HasPrivateKey);
```

```C#
Assert.NonNull(loaded.GetECDsaPrivateKey());
```
?

Reply:
`Assert.NonNull(loaded.GetECDsaPrivateKey());`

Makes sense to me, and implies `HasPrivateKey` is true.